### PR TITLE
teleport: annotations for ProxyV2 and MaxConnections

### PR DESCRIPTION
**Pull request by bpdohall**
This PR brings forward PR #4 where we add support for a couple of custom annotations on TLSRoutes:
1. Configure the route's transport to support Proxy protocol v2 connections to the backend.
2. Configure the route's cluster to use a non default setting for circuit breaker MaxConnections.

Migration to equivalent capabilities in upstream project: 
- MaxConnections is configurable via  [`BackendTrafficPolicy`](https://gateway.envoyproxy.io/docs/api/extension_types/#backendtrafficpolicy)`.spec.circuitBreaker.maxConections`
- ProxyV2 is configurable via [`BackendTrafficPolicy`](https://gateway.envoyproxy.io/docs/api/extension_types/#backendtrafficpolicy)`.spec.proxyProtocol.version`

### CODE REVIEW

File: internal/xds/translator/translator.go [hunk 1]
```diff
@@ -598,7 +598,7 @@ func (t *Translator) processTCPListenerXdsTranslation(
 		patchProxyProtocolFilter(xdsListener, tcpListener.EnableProxyProtocol)
 
 		for _, route := range tcpListener.Routes {
-			if err := processXdsCluster(tCtx, &TCPRouteTranslator{route}, &ExtraArgs{metrics: metrics}); err != nil {
+			if err := processXdsCluster(tCtx, &TCPRouteTranslator{route}, &ExtraArgs{metrics: metrics, circuitBreaker: route.CircuitBreaker}); err != nil {
 				errs = errors.Join(errs, err)
```
Comment (dboslee) [inline internal/xds/translator/translator.go]: I think this change is unnecessary. The cluster/proxy protocol config are in the route struct and will be taken from there.

File: internal/xds/translator/cluster.go [hunk 2]
```diff
@@ -695,9 +695,10 @@ func buildBackandConnectionBufferLimitBytes(bc *ir.BackendConnection) *wrappers.
 }
 
 type ExtraArgs struct {
-	metrics       *ir.Metrics
```
Comment (dboslee) [inline internal/xds/translator/cluster.go]: same as my other comment. I don't think you need to pass in the circuit breaker via extra args.

File: internal/xds/translator/translator.go (hunk 1 above)
Comment (bpdohall) [reply internal/xds/translator/translator.go]: you are correct - it tests out ok without passing the circuitbreaker around. [35c75cf](https://github.com/gravitational/gateway/pull/68/commits/35c75cf78a4f6e27e950c8fc969b440336f6f708)

File: internal/xds/translator/cluster.go (hunk 2 above)
Comment (dboslee) [reply internal/xds/translator/cluster.go]: can you revert this whole file? seems like the formatting is off. I would expect no diff here.

**Review by dboslee — APPROVED** [review]

**Pull request merged**